ISPN-10262 Deprecate custom interceptors configuration

danberindei opened the pull request:
https://issues.jboss.org/browse/ISPN-10262

ISPN-10262 Deprecate custom interceptors configuration

* Deprecate the custom interceptors configuration
* Warn at startup if custom interceptors are present
* Deprecate AdvancedCache.getAsyncInterceptorChain() 
* Replace interceptors configuration with direct AsyncInterceptorChain
  manipulation in all the modules.

ryanemerson commented on remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerImpl.java:
```diff
@@ -78,9 +66,6 @@ public ProtobufMetadataManagerImpl() {
    @Inject
    protected void init(EmbeddedCacheManager cacheManager, InternalCacheRegistry internalCacheRegistry) {
```
`internalCacheRegistry` no longer needed.

ryanemerson replied:
Can we just `@Inject EmbeddedCacheManager cacheManager` as a class variable?

ryanemerson commented on remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java:
```diff
@@ -225,4 +241,35 @@ public void cacheStarted(ComponentRegistry cr, String cacheName) {
    }
+
+
+   private void registerProtobufMetadataCache(InternalCacheRegistry internalCacheRegistry,
```
This method is never actually called.

danberindei replied:
Oops :)

ryanemerson commented:
@danberindei JCache seems to be blowing up

danberindei commented:
Fixed @ryanemerson. The server was not starting because I replaced uses of the custom interceptors configuration with direct manipulation of AsyncInterceptorChain and forgot to register the interceptors in the component registry.
